ddtrace/internal/encoding.py

```diff
@@ -156,6 +159,60 @@ def _decode_id_to_hex(hex_id):
         return int(hex_id, 16)
 
 
+class CIVisibilityEncoderV01(BufferedEncoder):
+    content_type = "application/msgpack"
+    ALLOWED_METADATA_KEYS = ("language", "library_version", "runtime-id", "env")
+
+    def __init__(self, *args):
+        super(CIVisibilityEncoderV01, self).__init__()
+        self._lock = threading.RLock()
+        self._init_buffer()
+
+    def __len__(self):
+        with self._lock:
+            return len(self.buffer)
+
+    def set_metadata(self, metadata):
+        self._metadata = metadata or dict()
+
+    def _init_buffer(self):
+        with self._lock:
+            self.buffer = []
+
+    def put(self, spans):
+        with self._lock:
+            self.buffer.append(spans)
+
+    def encode_traces(self, traces):
+        return self._build_payload(traces=traces)
+
+    def encode(self):
+        with self._lock:
+            payload = self._build_payload(self.buffer)
+            self._init_buffer()
+            return payload
+
+    def _build_payload(self, traces):
+        normalized_spans = [CIVisibilityEncoderV01._convert_span(span) for trace in traces for span in trace]
+        self._metadata = {k: v for k, v in self._metadata.items() if k in self.ALLOWED_METADATA_KEYS}
+        return msgpack_packb({"version": 1, "metadata": {"*": self._metadata}, "events": normalized_spans})
+
+    @staticmethod
+    def _convert_span(span):
+        # type: (Span) -> Dict[str, Any]
+        sp = JSONEncoderV2._convert_span(span)
+        sp["trace_id"] = JSONEncoderV2._encode_id_to_hex(span.trace_id)
```

Comment:
We should verify that the ci visibility intake supports decoding 128 bit trace ids. If 128 bit trace ids are not supported, we should use this trace id variant:

```suggestion
        sp["trace_id"] = JSONEncoderV2._encode_id_to_hex(span._trace_id_64bits)
```

Reply:
@juan-fernandez @albertvaka can you confirm if the CI Visibility intake supports 128 bit trace IDs?

Reply:
It does not.